pandas/core/dtypes/base.py

```diff
@@ -192,7 +207,40 @@ def construct_array_type(cls):
         raise NotImplementedError
 
     @classmethod
-    def construct_from_string(cls, string: str):
+    def _validate_from_string(cls, string: str) -> Dict[str, Any]:
+        """
+        Validate string argument of cls.construct_from_string().
+
+        If subclass defines class attribute `_match`, returns a dictionary
+        containing all the named subgroups of the match keyed by the subgroup name.
+        Used for keyword arguments to class constructor.
+
+        Returns
+        -------
+        dict
+        """
+        if not isinstance(string, str):
+            raise TypeError(
+                f"'construct_from_string' expects a string, got {type(string)}"
+            )
+
+        if hasattr(cls, "_match"):
```

Comment:
what uses this?

Reply:
some repeated logic in pandas/core/dtypes/dtypes.py can be removed with this. will remove for now til needed. (will also be adding a test in the base extension tests for consistent error messages for wrong type to construct_from_string.)

I'll open a PR to break off changes to pandas/core/dtypes/dtypes.py without the deduplication for now.